torch/csrc/jit/script/compiler.cpp

```diff
@@ -1484,6 +1530,10 @@ struct to_ir {
 // support syntax sugar for x.foo(y, z) by allowing x.foo to return a
 // callable value that will resolve to foo(x, y, z) when called.
 std::shared_ptr<SugaredValue> SimpleValue::attr(SourceRange loc, Method & m, const std::string& field) {
+  // Allow method-style casts on Tensor types. e.g. x.int()
+  if (value->type()->isSubtypeOf(DynamicType::get()) && ScalarTypeCastValue::method_to_type_map().count(field)) {
+    return std::make_shared<ScalarTypeCastValue>(field, NamedValue(loc, "self", value));
```

Comment:
I don't see the need for ScalarTypeCaseValue. In each case it is just a builtin function with a different name than standard. This can be accomplished by just doing the name translation here using method_to_type_map, and then returning a BuiltinFunction.